src/plasmapy/simulation/particle_tracker/particle_tracker.py

```diff
@@ -546,6 +586,161 @@ def inner_Bethe_stopping(v, n_e):
 
         self._log(f"Stopping module activated using the {method} method")
 
+        # TODO: create a method to validate that these quantities have shape ngrids
+        # TODO: is this signature too much?
+        # TODO: find a way to create a type for the signature of the Callable parameters
+        #  (preferably using only one expression)
+
+    @particle_input(require="isotope")
+    @validate_quantities
+    def add_scattering(
+        self,
+        target: ParticleListLike,
+        target_rho: u.Quantity[u.kg / u.m**3],
+        scatter_routine: Callable[
+            [
+                u.Quantity[u.rad] | None,  # Theta
+                u.Quantity[u.m / u.s],  # Speed
+                ParticleLike,  # Beam
+                ParticleLike,  # Target
+                u.Quantity[u.K],  # Electron Temperature
+                u.Quantity[1 / u.m**3],  # Electron Number Density
+            ],
+            u.Quantity[u.m**2 / u.sr]
+            | u.Quantity[
+                u.rad**2 / u.s
+            ],  # Differential cross-section or mean-square scatter rate in radians^2 per second
+        ],
+        scatter_routine_type: Literal["differential cross-section", "mean square rate"],
+    ):
+        r"""
+        Enable particle scattering by numeric integration of the provided
+        differential cross-section.
+
+        Parameters
+        ----------
+        target : `~plasmapy.particles.particle_collections.ParticleListLike`
+            A `~plasmapy.particles.particle_collections.ParticleListLike` representation of the target material(s).
+            The atomic mass of each particle must be specified (i.e. ``Al-27``).
+
+        scatter_routine : `callable`
```

Comment:
Is there a reason to support both of these? It does make the code a lot more complicated and double the number of cases to test. Let's discuss.